server/src/main/java/com/linecorp/centraldogma/server/internal/metadata/MetadataService.java

```diff
@@ -0,0 +1,853 @@
+        final Change<JsonNode> change =
+                Change.ofJsonPatch(METADATA_JSON,
+                                   asJsonArray(new TestAbsenceOperation(path),
+                                               new AddOperation(path, Jackson.valueToTree(newMember))));
+        final String commitSummary = "Add a member '" + newMember.id() + "' to the project " + projectName;
+        return metadataRepo.push(projectName, METADATA_REPO, author, commitSummary, change);
+    }
+
+    /**
+     * Removes the specified {@code member} from the {@link ProjectMetadata} of the specified
+     * {@code projectName}. It also removes permission of the specified {@code member} from every {@link Repo}.
+     */
+    public CompletionStage<Revision> removeMember(Author author, String projectName, User member) {
+        requireNonNull(author, "author");
+        requireNonNull(projectName, "projectName");
+        requireNonNull(member, "member");
+
+        final String commitSummary = "Remove the member '" + member.id() + "' from the project " + projectName;
+        return metadataRepo.push(
+                projectName, METADATA_REPO, author, commitSummary,
+                () -> fetchMetadata(projectName).thenApply(
+                        metadataWithRevision -> {
+                            final ImmutableList.Builder<JsonPatchOperation> patches = ImmutableList.builder();
+                            metadataWithRevision
+                                    .object().repos().values()
+                                    .stream().filter(r -> r.userPermission().containsKey(member.id()))
+                                    .forEach(r -> patches.add(new RemoveOperation(
+                                            jsonPointerOfUserPermission(r.name(), member.id()))));
+                            patches.add(new RemoveOperation(JsonPointer.compile("/members/" + member.id())));
+                            final Change<JsonNode> change =
+                                    Change.ofJsonPatch(METADATA_JSON, Jackson.valueToTree(patches.build()));
+                            return WithRevision.of(change, metadataWithRevision.revision());
+                        })
+        );
+    }
+
+    /**
+     * Updates a {@link ProjectRole} of the specified {@code member} in the specified {@code projectName}.
+     */
+    public CompletionStage<Revision> updateMemberRole(Author author, String projectName,
+                                                      User member, ProjectRole projectRole) {
+        requireNonNull(author, "author");
+        requireNonNull(projectName, "projectName");
+        requireNonNull(member, "member");
+        requireNonNull(projectRole, "projectRole");
+
+        final Change<JsonNode> change = Change.ofJsonPatch(
+                METADATA_JSON,
+                new ReplaceOperation(JsonPointer.compile("/members/" + member.id() + "/role"),
```

Comment:
Perhaps we should put some validation on its `id` property when a `User` is created so we do not have to worry about potential security issues?